pages/logout.tsx

```diff
@@ -1,21 +1,38 @@
 import { useEffect } from 'react'
+import { useRouter } from 'next/router'
 import { magic } from 'utils/magic'
-import Router from 'next/router'
 import Loader from 'components/Loader'
 import { encode as btoa } from 'base-64'
+import { LoginAware, STATUS } from 'hooks/useLogin'
 
-export default function Logout() {
+type LogoutProps = {
+  loginContext: LoginAware
+}
+
+export default function Logout({ loginContext }: LogoutProps) {
+  const $router = useRouter()
   useEffect(() => {
     const sayGoodbye = async () => {
-      if (!magic) return false
-      await magic.user.logout()
-      Router.push(
-        (await magic.user.isLoggedIn())
-          ? `/leaderboard`
-          : `/login?toast=${btoa('You have been logged out.')}`
-      )
+      // eslint-disable-next-line no-console
+      console.log('logging out')
+      if (magic && magic.user && magic.user.logout) {
```

Comment:
> If magic is defined, magic.user and magic.user.logout will always be defined according to the TS types so it's not necessary to check them.
Agreed, but I wrote this because TS wouldn't stop fighting me on it. Happy to try to revisit but I remember this specifically being the case.